lib/search/cubit/search_cubit.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:fluttercon/common/data/enums/search_result_type.dart';
+import 'package:fluttercon/common/data/models/failure.dart';
+import 'package:fluttercon/common/data/models/search_result.dart';
+import 'package:fluttercon/common/repository/db_repository.dart';
+import 'package:fluttercon/search/cubit/search_state.dart';
+
+class SearchCubit extends Cubit<SearchState> {
+  SearchCubit(
+    this._dbRepository,
+  ) : super(const SearchState.initial());
+
+  final DBRepository _dbRepository;
+
+  Future<void> search(String query) async {
+    emit(const SearchState.loading());
+    try {
+      final sessions = await _dbRepository.searchSessions(query);
+      final speakers = await _dbRepository.searchSpeakers(query);
+      final individualOrganizers =
+          await _dbRepository.searchIndividualOrganisers(query);
+
+      final results = [
+        ...sessions.map(
+          (session) => SearchResult(
+            id: session.serverId.toString(),
+            title: session.title,
+            subtitle: 'Session',
+            imageUrl: session.sessionImage,
+            type: SearchResultType.session,
+            session: session,
+          ),
+        ),
+        ...speakers.map(
+          (speaker) => SearchResult(
+            id: speaker.id.toString(),
+            title: speaker.name,
+            subtitle: speaker.tagline ?? '',
+            imageUrl: speaker.avatar,
+            type: SearchResultType.speaker,
+            speaker: speaker,
+          ),
+        ),
+        ...individualOrganizers.map(
+          (organizer) => SearchResult(
+            id: organizer.id.toString(),
```

Comment:
The `id:` attribute here doesn't seem to be in use anywhere. You can shave it.

Reply:
The id field is a required attribute in the SearchResult class, reflecting its importance in uniquely identifying each result. Retaining it in the search logic ensures consistency with the class definition and maintains the ability to distinguish between individual results throughout the application.